contracts/cw-native-staked-balance-voting/src/contract.rs

```diff
@@ -0,0 +1,287 @@
+    )?;
+
+    match config.unstaking_duration {
+        None => {
+            let msg = CosmosMsg::Bank(BankMsg::Send {
+                to_address: info.sender.to_string(),
+                amount: coins(amount.u128(), config.denom),
+            });
+            Ok(Response::new()
+                .add_message(msg)
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", amount)
+                .add_attribute("claim_duration", "None"))
+        }
+        Some(duration) => {
+            let outstanding_claims = CLAIMS.query_claims(deps.as_ref(), &info.sender)?.claims;
+            if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            CLAIMS.create_claim(
+                deps.storage,
+                &info.sender,
+                amount,
+                duration.after(&env.block),
+            )?;
+            Ok(Response::new()
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", amount)
+                .add_attribute("claim_duration", format!("{}", duration)))
+        }
+    }
+}
+
+pub fn execute_update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_owner: Option<String>,
+    new_manager: Option<String>,
+    duration: Option<Duration>,
+) -> Result<Response, ContractError> {
+    let new_owner = new_owner
+        .map(|new_owner| deps.api.addr_validate(&new_owner))
+        .transpose()?;
+    let new_manager = new_manager
+        .map(|new_manager| deps.api.addr_validate(&new_manager))
+        .transpose()?;
+    let mut config: Config = CONFIG.load(deps.storage)?;
+    if Some(info.sender.clone()) != config.owner && Some(info.sender.clone()) != config.manager {
+        return Err(ContractError::Unauthorized {});
+    };
+    if Some(info.sender) != config.owner && new_owner != config.owner {
+        return Err(ContractError::OnlyOwnerCanChangeOwner {});
+    };
+
+    config.owner = new_owner;
+    config.manager = new_manager;
+
+    config.unstaking_duration = duration;
+
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new()
+        .add_attribute("action", "update_config")
+        .add_attribute(
+            "owner",
+            config
+                .owner
+                .map(|a| a.to_string())
+                .unwrap_or_else(|| "None".to_string()),
+        )
+        .add_attribute(
+            "manager",
+            config
+                .manager
+                .map(|a| a.to_string())
+                .unwrap_or_else(|| "None".to_string()),
+        ))
+}
+
+pub fn execute_claim(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let release = CLAIMS.claim_tokens(deps.storage, &info.sender, &env.block, None)?;
```

Comment:
should we implement the cap via config, or leave it for now?

Reply:
leave for now, this approach has been fine for cw20-stake on mainnet